src/main/java/org/takes/facets/auth/social/LinkedinProfile.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.facets.auth.social;
+
+import com.jcabi.http.request.JdkRequest;
+import com.jcabi.http.response.JsonResponse;
+import com.jcabi.http.response.RestResponse;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Iterator;
+import javax.json.JsonObject;
+import org.takes.misc.Href;
+
+/**
+ * LinkedIn member profile in json format.
+ *
+ * <p>The class is immutable and thread-safe.
+ * @author Igor Khvostenkov (ikhvostenkov@gmail.com)
+ * @version $Id$
+ * @since 0.14.4
+ * @checkstyle MultipleStringLiteralsCheck (500 lines)
+ */
+final class LinkedinProfile implements MemberProfileJson {
+    /**
+     * App name.
+     */
+    private final transient String app;
+
+    /**
+     * Key.
+     */
+    private final transient String key;
+
+    /**
+     * Ctor.
+     * @param sapp Social network app
+     * @param skey Social network key
+     */
+    public LinkedinProfile(final String sapp, final String skey) {
```

Comment:
no need to make it public, since nobody except package neighbors can see it. you can configure your IDE and it will show this kind of "warning" to you. or just use my [settings.jar](http://img.teamed.io/settings.jar) for IntelliJ